data-prepper-plugins/http-sink/src/main/java/org/opensearch/dataprepper/plugins/sink/http/configuration/AuthenticationOptions.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.dataprepper.plugins.sink.http.configuration;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public class AuthenticationOptions {
+
+    @JsonProperty("http_basic")
+    private BasicAuthCredentials httpBasic;
+
+    @JsonProperty("bearer_token")
```

Comment:
Requirements also mentions mTLS. Are we supporting mTLS as well?

Reply:
Can you be more specific how it's supported: What would be the configuration look like? Are we going to have a separate auth option for mTLS?

Reply:
@oeyh ,In this plugin, we are supporting ssl/mtls with unauthenticated/http_basic/bearer token. If ssl_certificate is configured in both sender and receiver end, then with same configuration as ssl, given below there will be mutual authentication(mtls) too. There is no seperate auth option for mtls in the pipeline

configuration is 
ssl : true
ssl_certificate_file : <fullpath/httpsink.crt file>

Reply:
In the configuration for ssl, we are keeping ssl_certificate_file , ssl_key_file, The reason we are keeping ssl_key_file too, is because we are using a common class, FileCertificateProvider where in its constructor both certificate/key is validated as not null.